keyboards/kinesis/keymaps/kzar/keymap.c

```diff
@@ -0,0 +1,394 @@
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+                    _______, _______, _______, _______,
+           // Right Thumb
+           KC_RALT, _______,
+           _______,
+           _______, _______, _______
+    ),
+
+[_KEYPAD] = LAYOUT (
+           // Left Hand
+           _______, _______, _______, _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+                    _______, KC_INS,  _______, _______,
+           // Left Thumb
+                    _______, _______,
+                             _______,
+           _______, _______, _______,
+
+           // Right Hand
+           _______,    _______, _______, _______, KC__MUTE, KC__VOLDOWN, KC__VOLUP, _______, _______,
+           _______, KC_NLCK, KC_PEQL, KC_PSLS, KC_PAST,  _______,
+           _______, KC_P7,   KC_P8,   KC_P9,   KC_PMNS,  _______,
+           _______, KC_P4,   KC_P5,   KC_P6,   KC_PPLS,  _______,
+           _______, KC_P1,   KC_P2,   KC_P3,   KC_PENT,  _______,
+                    _______, _______, KC_PDOT, KC_PENT,
+           // Right Thumb
+           _______, _______,
+           _______,
+           _______, _______, KC_P0
+    ),
+
+[_PROGRAM] = LAYOUT (
+           // Left Hand
+           STATUS,  _______, _______, QWERTY,   DVORAK,  MAC,     PC,     WIN,     _______,
+           _______, _______, _______, _______ , _______, _______,
+           _______, _______, _______, _______ , _______, _______,
+           _______, _______, _______, _______ , _______, _______,
+           _______, _______, _______, _______ , _______, _______,
+                    _______, _______, _______ , _______,
+           // Left Thumb
+                    _______, _______,
+                             _______,
+           _______, _______, _______,
+
+           // Right Hand
+           RESET,   _______, _______, _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+           _______, _______, _______, _______, _______, _______,
+                    _______, _______, _______, _______,
+           // Right Thumb
+           _______, _______,
+           _______,
+           _______, _______, _______
+    )
+
+};
+
+void matrix_init_user(void) {
+
+}
+
+void matrix_scan_user(void) {
+    if (layer_state_is(_KEYPAD))
+        ledTurnOn(LED_KEYPAD);
+}
```

Comment:
Use `layer_state_set_user()` for this.

Reply:
I had a try at doing that, something like this:

```c
layer_state_t layer_state_set_user(layer_state_t state) {
  rgblight_set_layer_state(LED_KEYPAD, layer_state_cmp(state, _KEYPAD));
  return state;
}
```

But I got the error that `rgblight_set_layer_state` isn't defined. I then tried adding `RGBLIGHT_ENABLE = yes` to my rules file, but that led to the error that `RGBLED_NUM` is not defined in `quantum/rgblight.h`. Could you advise?

Reply:
Not sure where you got the rgblight stuff from...
```c
layer_state_t layer_state_set_user(layer_state_t state) {
    writePin(LED_KEYPAD, layer_state_cmp(state, _KEYPAD));
    return state;
}
```

Reply:
Thanks, that example compiles OK. Problem is it doesn't seem to trigger the keypad LED, bear with me I'm trying to figure out what's going on.

Reply:
Even if I simply call `writePinLow(LED_KEYPAD);` inside the `layer_state_set_user` function the keypad LED will not switch on. I'm wondering if `layer_state_set_user` is getting called at all...

Reply:
In fact, it seems like we might need to do this in `matrx_scan_user`. I believe `layer_state_set_user` is only called when the keymap is changed, but it seems that we need to keep setting the PIN low for as long as we want the light to stay on. What do you think?

Reply:
I did try inverting that (since we need to set the pin low to get the LED on), but no dice. That's what made me try simply calling `writePinLow(LED_KEYPAD);`, to see if I could force the LED to come on at all.

I'm still stuck here, with this code the LED stays off even after I have changed layer:

```
void keyboard_post_init_user(void) {
    setPinOutput(LED_KEYPAD);
}

layer_state_t layer_state_set_user(layer_state_t state) {
    writePinLow(LED_KEYPAD);
    return state;
}
```

It's like something keeps setting the pin high, which means we need to keep setting it low again to get the LED to turn on. I wonder why we can't just use `matrix_scan_user`? I guess it's less efficient to keep setting the PIN once when the layer changes.

Reply:
Ah, I see E26 is defined as the Compose LED in kint36/config.h. That'll be what is forcing it high. I guess you will need a config.h after all:

```c
#pragma once

#undef LED_COMPOSE_PIN
```

Reply:
Woohoo, that works thanks! Done.